libbeat/logp/file_rotator.go

```diff
@@ -56,6 +57,10 @@ func (rotator *FileRotator) CheckIfConfigSane() error {
 	if *rotator.KeepFiles < 2 || *rotator.KeepFiles >= RotatorMaxFiles {
 		return fmt.Errorf("The number of files to keep should be between 2 and %d", RotatorMaxFiles-1)
 	}
+
+	if rotator.Permissions != nil && (*rotator.Permissions > 7777) {
+		return fmt.Errorf("The permissions mask %d is invalid", *rotator.Permissions)
```

Comment:
Error strings should not be capitalized. https://github.com/golang/go/wiki/CodeReviewComments#error-strings